packages/runtime/src/next/draft-mode/draft-mode-script.tsx

```diff
@@ -1,12 +1,13 @@
-import { draftMode } from 'next/headers'
+import { use } from 'react'
 import { ActionTypes } from '../../react'
+import { draftMode } from '../lib/next'
 
 type Props = {
   appOrigin?: string
 }
 
 export function DraftModeScript({ appOrigin = 'https://app.makeswift.com' }: Props) {
-  const { isEnabled: isDraftModeEnabled } = draftMode()
+  const { isEnabled: isDraftModeEnabled } = use(draftMode())
```

Comment:
This might be problematic if used in a root layout marked with `use client`. Not sure why anybody would do that but if somebody does, I've seen React complain about using `use` with a non-cached promise (i.e., promise created on render). It seems that we probably might want to just go ahead and make this a RSC?

Reply:
Yep, was revising this just as you commented.